src/main/java/bio/terra/service/configuration/ConfigFault.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.service.configuration;
+
+import bio.terra.app.controller.exception.ValidationException;
+import bio.terra.model.ConfigFaultModel;
+import bio.terra.model.ConfigModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// Sub-classes must implement:
+// get()
+// setFaultConfig()
+public abstract class ConfigFault extends ConfigBase {
+    private final Logger logger = LoggerFactory.getLogger(ConfigFault.class);
+
+    private final ConfigFaultModel.FaultTypeEnum faultType;
+    private final boolean originalEnabled;
+    private boolean enabled;
+
+    // -- Instance methods --
+    ConfigFault(ConfigEnum configEnum,
+                ConfigFaultModel.FaultTypeEnum faultType,
+                boolean enabled) {
+        super(ConfigModel.ConfigTypeEnum.FAULT, configEnum);
+        this.faultType = faultType;
+        this.originalEnabled = enabled;
+        this.enabled = enabled;
+    }
+
+    @Override
+    synchronized void reset() {
+        enabled = originalEnabled;
+        resetFaultConfig();
+    }
+
+    boolean isEnabled() {
```

Comment:
findbugs did not complain - and it did complain elsewhere... I think since it is a read of an atomic sized thing, it doesn't need to be interlocked. I'd rather not force an interlock, since the method is frequently called in the no-faults-on mode.